nano/node/bootstrap.cpp

```diff
@@ -1739,20 +1743,10 @@ void nano::bootstrap_server::receive_header_action (boost::system::error_code co
 			{
 				case nano::message_type::bulk_pull:
 				{
-					uint32_t extended_size;
 					node->stats.inc (nano::stat::type::bootstrap, nano::stat::detail::bulk_pull, nano::stat::dir::in);
 
-					if (header.bulk_pull_is_count_present ())
-					{
-						extended_size = nano::bulk_pull::extended_parameters_size;
-					}
-					else
-					{
-						extended_size = 0;
-					}
-
 					auto this_l (shared_from_this ());
-					socket->async_read (receive_buffer, sizeof (nano::uint256_union) + sizeof (nano::uint256_union) + extended_size, [this_l, header](boost::system::error_code const & ec, size_t size_a) {
+					socket->async_read (receive_buffer, nano::bulk_pull::size (header.bulk_pull_is_count_present ()), [this_l, header](boost::system::error_code const & ec, size_t size_a) {
```

Comment:
Why not make the header size a method of the header object ?

Reply:
extended_parameters_size / count_present_flag was already part of the bulk_pull class.

Reply:
added `payload_length_bytes`